cmd/account/mgmt/account-unassign.go

```diff
@@ -0,0 +1,370 @@
+
+func (o *accountUnassignOptions) listAccountsFromUser(user string) ([]string, error) {
+
+	inputFilterTag := &resourcegroupstaggingapi.GetResourcesInput{
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key: aws.String("owner"),
+				Values: []*string{
+					aws.String(user),
+				},
+			},
+		},
+	}
+	accounts, err := o.awsClient.GetResources(inputFilterTag)
+	if err != nil {
+		return []string{}, err
+	}
+
+	if len(accounts.ResourceTagMappingList) == 0 {
+		return []string{}, ErrNoAccountsForUser
+	}
+
+	var accountIdList []string
+	// Get last 12 digits of ResourceARN and append it to account list
+	for _, a := range accounts.ResourceTagMappingList {
+		accountIdList = append(accountIdList, (*a.ResourceARN)[len(*a.ResourceARN)-12:])
+	}
+	if err != nil {
+		return []string{}, err
+	}
+	return accountIdList, nil
+}
+
+func (o *accountUnassignOptions) deleteLoginProfile(user string) error {
+
+	inputDeleteLogin := &iam.DeleteLoginProfileInput{
+		UserName: &user,
+	}
+	_, err := o.awsClient.DeleteLoginProfile(inputDeleteLogin)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *accountUnassignOptions) deleteAccessKeys(user string) error {
+
+	inputListAccessKeys := &iam.ListAccessKeysInput{
+		UserName: &user,
+	}
+	accessKeys, err := o.awsClient.ListAccessKeys(inputListAccessKeys)
+	if err != nil {
+		return err
+	}
+
+	for _, m := range accessKeys.AccessKeyMetadata {
+
+		inputDelKey := &iam.DeleteAccessKeyInput{
+			AccessKeyId: m.AccessKeyId,
+			UserName:    &user,
+		}
+		_, err = o.awsClient.DeleteAccessKey(inputDelKey)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *accountUnassignOptions) deleteSigningCert(user string) error {
+
+	inputListCert := &iam.ListSigningCertificatesInput{
+		UserName: &user,
+	}
+	cert, err := o.awsClient.ListSigningCertificates(inputListCert)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range cert.Certificates {
+		inputDelCert := &iam.DeleteSigningCertificateInput{
+			CertificateId: c.CertificateId,
+			UserName:      &user,
+		}
+		_, err = o.awsClient.DeleteSigningCertificate(inputDelCert)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *accountUnassignOptions) deletePolicies(user string) error {
+	inputListPolicies := &iam.ListUserPoliciesInput{
+		UserName: &user,
+	}
+	policies, err := o.awsClient.ListUserPolicies(inputListPolicies)
+	if err != nil {
+		return err
+	}
+
+	for _, p := range policies.PolicyNames {
+		inputDelPolicies := &iam.DeleteUserPolicyInput{
+			PolicyName: p,
+			UserName:   &user,
+		}
+		_, err = o.awsClient.DeleteUserPolicy(inputDelPolicies)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *accountUnassignOptions) deleteAttachedPolicies(user string) error {
+	inputListAttachedPol := &iam.ListAttachedUserPoliciesInput{
+		UserName: &user,
+	}
+	attachedPol, err := o.awsClient.ListAttachedUserPolicies(inputListAttachedPol)
+	if err != nil {
```

Comment:
You forgot to test this situation. Please add this as part of your unit-tests ;)